core/src/main/scala/kafka/utils/Logging.scala

```diff
@@ -17,16 +17,35 @@
 
 package kafka.utils
 
-import org.apache.log4j.Logger
+import org.slf4j.{LoggerFactory, MarkerFactory}
+
+import scala.util.{Failure, Success, Try}
+
+object Log4jControllerRegistration {
+  private val logger = LoggerFactory.getLogger(this.getClass.getName)
+
+  val instance = Try {
+    val log4jController = Class.forName("kafka.utils.Log4jController").asInstanceOf[Class[Object]]
+    val instance = log4jController.newInstance()
+    CoreUtils.registerMBean(instance, "kafka:type=kafka.Log4jController")
+    instance
+  }
+
+  instance match {
+    case Success(v) => logger.info("Registered kafka:type=kafka.Log4jController MBean")
+    case Failure(v) => logger.info("Couldn't register kafka:type=kafka.Log4jController MBean")
+  }
+}
 
 trait Logging {
   val loggerName = this.getClass.getName
-  lazy val logger = Logger.getLogger(loggerName)
+  lazy val logger = LoggerFactory.getLogger(loggerName)
 
   protected var logIdent: String = null
 
-  // Force initialization to register Log4jControllerMBean
-  private val log4jController = Log4jController
+  val fatalMarker = MarkerFactory.getMarker("FATAL")
```

Comment:
Like this?
`logger.error(Log4jControllerRegistration.fatalMarker, msgWithLogIdent(msg))`